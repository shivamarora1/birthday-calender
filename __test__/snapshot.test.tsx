import { render } from "@testing-library/react";
import Day from "../app/ui/day";
import AuthWall from "../app/ui/auth";
import Loader from "../app/ui/loader";

it("day renders correctly", () => {
  const tree = render(
    <Day
      date={new Date(2023, 1, 1)}
      eventsParam={["Event-1", "Event-2"]}
      showModal={(day, month) => {
        return `Day is ${day} and Month is ${month}`;
      }}
    />
  );
  expect(tree).toMatchSnapshot();
});

it("auth renders correctly", () => {
  const tree = render(<AuthWall handleVerifyAuth={(passcode: string) => {}} />);
  expect(tree).toMatchSnapshot();
});

it("show loader renders correctly", () => {
  const tree = render(<Loader showLoader={true} />);
  expect(tree).toMatchSnapshot();
});

it("hide loader renders correctly", () => {
  const tree = render(<Loader showLoader={false} />);
  expect(tree).toMatchSnapshot();
});
