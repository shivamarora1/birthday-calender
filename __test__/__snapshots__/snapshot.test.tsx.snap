// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full absolute top-0 bg-black bg-opacity-50 flex justify-center items-center"
      >
        <div
          class="w-6/6 h-1/4 sm:w-3/6 sm:h-1/6 lg:w-1/6 lg:h-1/6 border p-2 bg-white rounded-xl"
        >
          <div>
            <label
              class="block mx-1 text-lg font-medium text-black"
            >
              Enter secret code to proceed.
            </label>
            <input
              class="bg-gray-50 border border-gray-300 text-black text-sm rounded-lg focus:border-blue-500 black w-full p-2.5"
              id="passcode"
              placeholder="Enter passcode"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="mt-5 flex text-center justify-center items-center"
          >
            <button
              class="bg-blue-500 center w-20 hover-bg-blue-700 text-white font-bold py-2 px-4 rounded mr-2"
            >
              Ok
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full absolute top-0 bg-black bg-opacity-50 flex justify-center items-center"
    >
      <div
        class="w-6/6 h-1/4 sm:w-3/6 sm:h-1/6 lg:w-1/6 lg:h-1/6 border p-2 bg-white rounded-xl"
      >
        <div>
          <label
            class="block mx-1 text-lg font-medium text-black"
          >
            Enter secret code to proceed.
          </label>
          <input
            class="bg-gray-50 border border-gray-300 text-black text-sm rounded-lg focus:border-blue-500 black w-full p-2.5"
            id="passcode"
            placeholder="Enter passcode"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="mt-5 flex text-center justify-center items-center"
        >
          <button
            class="bg-blue-500 center w-20 hover-bg-blue-700 text-white font-bold py-2 px-4 rounded mr-2"
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`day renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="block border-l border-b border-gray-600 w-48 h-44 text-center 
        border-t 
         
        
        rounded-tl-lg
        
        
        
        "
      >
        <div
          class="text-xs"
        >
          Wednesday
        </div>
        <div
          class="text-sm font-medium"
        >
          1
        </div>
        <div
          class="text-sm mt-1 mb-1 ml-2"
        >
          <div
            class="mb-0.5 rounded-l bg-gray-400 text-white"
          >
            Event-1
          </div>
          <div
            class="mb-0.5 rounded-l bg-gray-300 text-black"
          >
            Event-2
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="block border-l border-b border-gray-600 w-48 h-44 text-center 
        border-t 
         
        
        rounded-tl-lg
        
        
        
        "
    >
      <div
        class="text-xs"
      >
        Wednesday
      </div>
      <div
        class="text-sm font-medium"
      >
        1
      </div>
      <div
        class="text-sm mt-1 mb-1 ml-2"
      >
        <div
          class="mb-0.5 rounded-l bg-gray-400 text-white"
        >
          Event-1
        </div>
        <div
          class="mb-0.5 rounded-l bg-gray-300 text-black"
        >
          Event-2
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`hide loader renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`show loader renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full text-2xl absolute font-bold top-0 bg-black bg-opacity-50 flex justify-center items-center"
      >
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full text-2xl absolute font-bold top-0 bg-black bg-opacity-50 flex justify-center items-center"
    >
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
